src/main/java/uk/gov/homeoffice/digital/sas/timecard/validators/timeentry/TimeEntryValidator.java

```diff
@@ -30,6 +35,102 @@ public boolean isValid(Object value, ConstraintValidatorContext context) {
         timeEntry.getActualEndTime());
 
     session.setHibernateFlushMode(FlushMode.AUTO);
-    return timeEntryClashes.isEmpty();
+    if (!timeEntryClashes.isEmpty()) {
+      ClashingProperty clashingProperty = getClashingProperty(timeEntry, timeEntryClashes);
+      var payload = timeEntryClashes.stream().map(this::transformTimeEntry)
+          .collect(Collectors.toCollection(ArrayList::new));
+
+      HibernateConstraintValidatorContext hibernateContext =
+          context.unwrap(HibernateConstraintValidatorContext.class);
+
+      hibernateContext.disableDefaultConstraintViolation();
+      hibernateContext.withDynamicPayload(payload);
+      hibernateContext
+          .buildConstraintViolationWithTemplate(
+              "Time periods must not overlap with another time period")
+          .addPropertyNode(clashingProperty.toString())
+          .addConstraintViolation();
+      return false;
+    }
+    return true;
+  }
+
+  private JSONObject transformTimeEntry(TimeEntry timeEntry) {
+    var result = new JSONObject();
+    result.put("startTime", timeEntry.getActualStartTime());
+    result.put("endTime", timeEntry.getActualEndTime());
+    result.put("timePeriodTypeId", timeEntry.getTimePeriodTypeId());
+    return result;
+  }
+
+  private ClashingProperty getClashingProperty(
+      TimeEntry timeEntry, List<TimeEntry> timeEntryClashes) {
+    var startTimeClash = false;
+    var endTimeClash = false;
+
+    for (TimeEntry timeEntryClash : timeEntryClashes) {
+      if (timeEntry.getActualStartTime().equals(timeEntryClash.getActualStartTime())) {
+        startTimeClash = true;
```

Comment:
I can see you have discussed combining this with Daleel above. But this is coming up as critical code smell. Would suggest combining, To avoid building in tech debt.